### PR TITLE
Add Flow.UseExecutor

### PR DESCRIPTION
**Pull request by pellared**
## Why

Fixes https://github.com/goyek/goyek/issues/416

## What

- Add `Flow.UseExecutor` method to support flow execution interception using middlewares.

## Checklist

- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated.
- [x] The code changes follow [Effective Go](https://golang.org/doc/effective_go).
- [x] The code changes follow [CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments).
- [x] The code changes are covered by tests.

### CODE REVIEW

File: executor.go [hunk 1]
```diff
@@ -5,30 +5,49 @@ import (
...
+	ExecutorMiddleware func(Executor) Executor
+
+	executor struct {
+		defined     map[string]*taskSnapshot
+		middlewares []Middleware
+	}
+)
 
 // Execute runs provided tasks and all their dependencies.
 // Each task is executed at most once.
```
Comment (ccoVeille) [inline executor.go:32]: I would add this method is there to satisfy/implements Executor interface

File: flow.go [hunk 2]
```diff
@@ -278,12 +276,27 @@ func Use(middlewares ...Middleware) {
 func (f *Flow) Use(middlewares ...Middleware) {
 	for _, m := range middlewares {
 		if m == nil {
-			panic("middleware cannot be nil")
+			panic("middlewares cannot be nil")
```
Comment (ccoVeille) [inline flow.go:279]: Why using plural? You are only checking one middle ware at a time

File: executor.go (hunk 1 above)
Comment (pellared) [reply executor.go:32]: Not needed as other code would not compile.

File: flow.go (hunk 2 above)
Comment (pellared) [reply flow.go:279]: Thanks https://github.com/goyek/goyek/pull/419

**Pull request merged**